### PR TITLE
Create Georef_A_12d.ifc

### PR DESCRIPTION
**Pull request by AdamMackellar12d**
## File

Basic ifc file with a single point in map coordinates (case: Georef_A)
Inclusion of Project Units including a plane angle measure, Actor attached to Project, Site/Facility, ArchElementType, Document, Dataset, Library, and Property Set Template.

## Covered Concept Templates - Usages Included

1. IfcRelDeclares for IfcProject - Sprint 1.1
2. IfcActor associated with IfcProject - Sprint 1.1
3. IfcUnitAssignment - Sprint 1.1
4. IfcGeometricContext (3D) - Sprint 1.1
5. IfcClassification - Sprint 1.2
6. IfcSite and IfcFacility - Sprint 1.2
7. IfcProjectedCRS with associated IfcMapConversion - Sprint 1.2
8. IfcDocumentInformation - Sprint 1.3
9. IfcDatasetInformation - Sprint 1.3
10. IfcLibraryInformation - Sprint 1.3
11. IfcArchElementType - Sprint 1.3
12. IfcPropertySetTemplate with associated IfcPropertyTemplate - Sprint 1.3

## Miscellaneous

*Please use this area to convey anything else.*

### CODE REVIEW

**Comment by pjanck** [comment]
@AdamMackellar12d Thank you for the submitted file. While the checker marked your file as correct in the first run, it did not stop at an exception (my bad). This has been corrected and now the error messages are correctly shown in [check summary](https://github.com/bSI-InfraRoom/IFC-Tunnel-Deployment/actions/runs/4412680104).

File: files/12dSolutions/Georef_A_12d.ifc [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+#17 = IFCDIRECTION((0., 0., 1.));
+#18 = IFCDIRECTION((1., 0., 0.));
+#19 = IFCDIRECTION((0., 1.));
+#20 = IFCPROJECTEDCRS('EPSG:3857', $, 'World Geodetic System 1984 ensemble', $, $, $, #9);
+#21 = IFCMAPCONVERSION(#13, #20, 16833194.0548, -4009644.0757, 0., 1., 0., 1.);
+#22 = IFCRELDECLARES('1sZoCzuQf8ERSGp0i$7pFM', #2, '', '', #1, (#23, #32, #33));
+#23 = IFCACTOR('0P5Q94Z2D2Zu2JT8cenAl2', $, $, $, $, #4);
+#24 = IFCCLASSIFICATION('Uniclass 2015', '1', '23-04-2018', 'Uniclass', 'Unified construction classification', 'uniclass.thenbs.com', $);
+#25 = IFCRELASSOCIATESCLASSIFICATION('2TOKspu3bCjvFbDbmg1OkX', #2, $, $, (#1), #24);
+#26 = IFCDOCUMENTINFORMATION('project_plan', 'project_plan.pdf', $, 'https://github.com/bSI-InfraRoom/IFC-Tunnel-Deployment/blob/main/project_plan.pdf', $, $, $, $, $, $, $, $, 'application/pdf', $, $, $, $);
```
Comment (pjanck) [inline files/12dSolutions/Georef_A_12d.ifc:33]: An Editor (10th attribute) is missing.

File: files/12dSolutions/Georef_A_12d.ifc [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+#27 = IFCRELASSOCIATESDOCUMENT('1yt1CEMdbCSRKx$yN4Hr$z', #2, '', '', (#1), #26);
+#28 = IFCDATASETINFORMATION('Dataset-test', 'Dataset-test-xml', $, 'http://xxx.yyy.se/Datasets/Dataset-test.xml', $, $, $, $, $, $, $, $, 'text/xml', $, $, $, $, 'http://geosciml.org/geosciml/2.0/xsd/geosciml.xsd');
+#29 = IFCRELASSOCIATESDATASET('1Da8HG4NjD7PrRWACz8vIa', #2, '', '', (#1), #28);
+#30 = IFCLIBRARYINFORMATION('Example Library', '1.2.3', #3, '2023-03-10T13:05:38', 'http://xxx.yyy.se/Libraries/ExampleLibrary.ifc', $);
+#31 = IFCRELASSOCIATESLIBRARY('0X5yi7ymLDtAPHah_cGdyA', #2, '', '', (#1), #30);
+#32 = IFCARCHELEMENTTYPE('3l9$2hAxr6Hx$RitoS1PAI', #2, 'ArchType Example', $, $, $, $, $, 'MyArch', .USERDEFINED.);
+#33 = IFCPROPERTYSETTEMPLATE('30$TWUmhXCpQik7$iwa1Mi', #2, 'Pset_GeoObsBoreholeConstruction', $, .PSET_TYPEDRIVENOVERRIDE., 'IfcBorehole', (#34));
+#34 = IFCSIMPLEPROPERTYTEMPLATE('1lcv35HjrE1R$ee0lT6V6c', #2, 'DrillingDiameter', $, .P_SINGLEVALUE., $, $, $, $, $, $, $);
```
Comment (pjanck) [inline files/12dSolutions/Georef_A_12d.ifc:41]: The attribute `PrimaryMeasureType` shall be 'IfcPositiveLengthMeasure'.

**Review by pjanck — CHANGES_REQUESTED** [review]

File: files/12dSolutions/Georef_A_12d.ifc [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+#12 = IFCSIUNIT(*, .PLANEANGLEUNIT., $, .RADIAN.);
+#13 = IFCGEOMETRICREPRESENTATIONCONTEXT('3D', 'Model', 3, 1.E-6, #15, #19);
+#14 = IFCLOCALPLACEMENT($, #15);
+#15 = IFCAXIS2PLACEMENT3D(#16, #17, #18);
+#16 = IFCCARTESIANPOINT((0., 0., 0.));
+#17 = IFCDIRECTION((0., 0., 1.));
+#18 = IFCDIRECTION((1., 0., 0.));
+#19 = IFCDIRECTION((0., 1.));
+#20 = IFCPROJECTEDCRS('EPSG:3857', $, 'World Geodetic System 1984 ensemble', $, $, $, #9);
+#21 = IFCMAPCONVERSIONSCALED(#13, #20, 16833194.0548, -4009644.0757, 0., 1., 0., 1., 1., 1., 1.);
```
Comment (pjanck) [inline files/12dSolutions/Georef_A_12d.ifc:28]: Is this a specific point on the map, or are number chosen at random?

Also, I would love to see an Australian EPSG used in practice here, and not something that is [21km precise on ground](https://epsg.io/3857) :)

File: files/12dSolutions/Georef_A_12d.ifc [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+#35 = IFCSIMPLEPROPERTYTEMPLATE('1lcv35HjrE1R$ee0lT6V6c', #2, 'DrillingDiameter', $, .P_SINGLEVALUE., 'IfcPositiveLengthMeasure', $, $, $, $, $, $);
+#36 = IFCSITE('0hf1$o_8H5dQQ1kuoHYlXh', #2, 'Default Site', 'Description of Default Site', $, #14, $, $, .ELEMENT., $, $, $, $, $);
+#37 = IFCRELAGGREGATES('1wUGpEYtn14gG33sBzRt7A', #2, 'ProjectContainer', 'ProjectContainer for Sites', #1, (#36));
+#38 = IFCFACILITY('09i1GJI4X3J8ubPZYfcSmx', #2, 'base', 'Description of base', $, #14, $, $, .ELEMENT.);
+#39 = IFCRELAGGREGATES('2V4VdApF1BdPSvMLZkah4J', #2, 'SiteContainer', 'SiteContainer For All Models', #36, (#38));
+#40 = IFCRELCONTAINEDINSPATIALSTRUCTURE('2EXlqCgUPDjO6fvIKZmHrY', #2, 'FacilityContainer', 'FacilityContainer for Elements', (#41), #38);
+#41 = IFCBUILDINGELEMENTPROXY('2wTV1IXaf4Ev6UbKzFLN7o', #2, 'Super', $, 'Super', #14, #42, $, .NOTDEFINED.);
+#42 = IFCPRODUCTDEFINITIONSHAPE($, $, (#43));
+#43 = IFCSHAPEREPRESENTATION(#13, 'Body', 'Point', (#44));
+#44 = IFCCARTESIANPOINT((-16833184.0548, 4009649.0757, 0.));
```
Comment (pjanck) [inline files/12dSolutions/Georef_A_12d.ifc:51]: This point seems to negate almost the whole offset of the `IfcMapConversionScaled`, however in engineering coordinates (not mapped). Is this intentional?

**Review by pjanck — APPROVED** [review]
Since the examples are fictitious, there is technically nothing wrong - the comments below are more of informative nature. LFTM

File: files/12dSolutions/Georef_A_12d.ifc (hunk 4 above)
Comment (AdamMackellar12d) [reply files/12dSolutions/Georef_A_12d.ifc:51]: I was just using a random point for the purposes of getting our software to export a file, so the offset defined in IfcMapConversionScaled will be mostly negated as I just put a point close to the (0,0,0) before I setup the map conversion.

File: files/12dSolutions/Georef_A_12d.ifc (hunk 3 above)
Comment (AdamMackellar12d) [reply files/12dSolutions/Georef_A_12d.ifc:28]: I think that should be the Sydney Opera House.

**Review by larswik — APPROVED** [review]

**Comment by AdamMackellar12d** [comment]
This should only compliant up to sprint 1.3

File: files/12dSolutions/Georef_A_12d.ifc [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+#37 = IFCSIMPLEPROPERTYTEMPLATE('1lcv35HjrE1R$ee0lT6V6c', #2, 'DrillingDiameter', $, .P_SINGLEVALUE., 'IfcPositiveLengthMeasure', $, $, $, $, $, $);
+#38 = IFCRELAGGREGATES('3niR2H_xX4avsspSUfuF$r', #2, 'ProjectContainer', 'ProjectContainer for Sites', #1, (#39, #49));
+#39 = IFCTUNNEL('0Jd9SK1MLFwOa3VO4HUs_G', #2, 'Tunnel 01', $, $, #14, $, $, .ELEMENT., $);
+#40 = IFCRELAGGREGATES('37up2TQXD3WvrtWAE4PQU$', #2, 'Tunnel 01 Spatial Container', 'Tunnel 01 Container for Spatial Elements', #39, (#41));
+#41 = IFCTUNNELPART('04cmyVtbL7QOVIuSxhRxbG', #2, 'Tunnel Part 01', $, $, #14, $, $, .ELEMENT., $, $);
+#42 = IFCRELCONTAINEDINSPATIALSTRUCTURE('1EVwZ9jxnC8BIWMPnI$kP2', #2, 'Tunnel Part 01 Container', 'Tunnel Part 01 Container for Elements', (#43), #41);
+#43 = IFCBUILDINGELEMENTPROXY('2wTV1IXaf4Ev6UbKzFLN7o', #2, 'Super', $, 'Super', #14, #44, $, .NOTDEFINED.);
+#44 = IFCPRODUCTDEFINITIONSHAPE($, $, (#45));
+#45 = IFCSHAPEREPRESENTATION(#13, 'Body', 'Point', (#46));
+#46 = IFCCARTESIANPOINT((-16330482.0548, 10971354.0757, 0.));
```
Comment (pjanck) [inline files/12dSolutions/Georef_A_12d.ifc:53]: Looking at `#21 = IFCMAPCONVERSIONSCALED(#13, #20, 16833194.0548, -4009644.0757, 0., 1., 0., 1., 1., 1., 1.);` the coordinate values of the `IfcCartesianPoint` seem weird. Please double check with @LeeGregory12d.

**Review by pjanck — CHANGES_REQUESTED** [review]

File: files/12dSolutions/Georef_A_12d.ifc [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+#11 = IFCSIUNIT(*, .VOLUMEUNIT., $, .CUBIC_METRE.);
+#12 = IFCSIUNIT(*, .PLANEANGLEUNIT., $, .RADIAN.);
+#13 = IFCGEOMETRICREPRESENTATIONCONTEXT('3D', 'Model', 3, 1.E-6, #15, #19);
+#14 = IFCLOCALPLACEMENT($, #15);
+#15 = IFCAXIS2PLACEMENT3D(#16, #17, #18);
+#16 = IFCCARTESIANPOINT((0., 0., 0.));
+#17 = IFCDIRECTION((0., 0., 1.));
+#18 = IFCDIRECTION((1., 0., 0.));
+#19 = IFCDIRECTION((0., 1.));
+#20 = IFCPROJECTEDCRS('MGA2020 Zone 55 with AHD', 'HD: MGA2020, Zone 55 VD: AHD', 'EPSG:7855', 'EPSG:5711', 'UTM', '55', #9);
```
Comment (larswik) [inline files/12dSolutions/Georef_A_12d.ifc:27]: one of these shall be true:
IfcProjectedCRS.Name shall begin with EPSG: and be followed by 4 or 5 digits (actual numbers are irrelevant to the checker).
IfcProjectedCRS.HasWellKnownTextRepresentation.WellKnownText shall be non-empty.

**Review by larswik — CHANGES_REQUESTED** [review]
The name attribute should have the EPSG code.

**Comment by AdamMackellar12d** [comment]
This latest commit should only be valid up to sprint 2.2

**Review by pjanck — APPROVED** [review]
lftm

**Review by larswik — APPROVED** [review]
lftm

**Pull request merged**